engine/src/main/java/org/destinationsol/game/BeaconHandler.java

```diff
@@ -183,10 +183,10 @@ private void updateD(SolGame game) {
                 farDrawable = (FarDrawable) fo;
                 return;
             }
-            throw new AssertionError();
+            throw new AssertionError("The far object data in game is a FarDrawable object, does not contain 3 elements in drawables, or the first drawable isnt attackSprite");
```

Comment:
This exception is reached when *every single* far object data does not conform to these conditions, not just one.